File: 413-arithmetic-slices/413-arithmetic-slices.java
class Solution {
    public int numberOfArithmeticSlices(int[] nums) {
        int curr = 0;
        int ans = 0;
        if (nums.length < 3)
            return 0;
        for (int i = 2; i < nums.length ; i ++){
            if (nums[i] - nums[i - 1] == nums[i - 1] - nums[i - 2]){
                curr ++;
            } else{
                curr = 0;
            }
            ans = ans + curr ;
        }
        return ans ;
    }
}

File: 413-arithmetic-slices/413-arithmetic-slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        if len(nums) < 3:
            return 0
        curr = 0 ; ans = 0
        for i in range(2,len(nums)):
            if nums[i] - nums[i - 1] == nums [i - 1] - nums[ i - 2 ]:
                curr += 1
            else:
                curr = 0
            ans += curr
        return ans
            
        